### PR TITLE
Add a new ci-reporter tool to generate weekly CI Signal Reports

### PR DESCRIPTION
**Pull request by palnabarun**
#### What type of PR is this?

/kind documentation
/kind feature

#### What this PR does / why we need it:

There have been discussions around porting the existing CI Signal Report tooling (https://github.com/alenkacz/ci-signal-report) to this repository. This PR attempts to unblock the CI Signal team and move the same to k/release as a new program for now with further plans as discussed in https://kubernetes.slack.com/archives/C2C40FMNF/p1622718810193000 and how the CI Signal Team would want to develop the tooling.

### Open Questions

- **[RESOLVED - Stephen's comment below answers the question]** Who are the OWNERS of the `cmd/ci-reporter` directory? I have added the 1.23 CI Signal Team as approvers/reviewers for now. We may need folks who are around irrespective of the release cycles.
- Some new dependencies also got added with the code
  - `github.com/google/go-github/v34`
    - `github.com/google/go-github/v33` is being used in this repo already as an indirect dependency. We need to figure out if we really need both and see if we can use just one version. 
  - **[RESOLVED - We are not pulling that dependency anymore]** `github.com/kelseyhightower/envconfig`
    - This is being used only once to read the `GITHUB_AUTH_TOKEN` environment variable. I guess this can be removed since we are just reading one environment variable and `os.Getenv` can be used instead. 

#### Which issue(s) this PR fixes:

Fixes https://github.com/kubernetes/sig-release/issues/1586
Fixes https://github.com/kubernetes/org/issues/3044
Progresses on https://kubernetes.slack.com/archives/C2C40FMNF/p1622718810193000
Unblocks https://kubernetes.slack.com/archives/CN0K3TE2C/p1636398789023800?thread_ts=1636389405.015600&cid=CN0K3TE2C

#### Special notes for your reviewer:

Please have a look at the commits for attribution to the Authors who contributed to https://github.com/alenkacz/ci-signal-report. 

#### Does this PR introduce a user-facing change?

```release-note
Add a new ci-reporter tool to generate weekly CI Signal Reports
```

/area release-team
/priority critical-urgent 
/cc @encodeflush @calvh @leonardpahlke @rajula96reddy @RinkiyaKeDad @simrangupta234
/cc @kubernetes/release-engineering
cc: @kubernetes/release-team-leads

### CODE REVIEW

**Comment by palnabarun** [comment]
@mkorbi @leonardpahlke -- I found multiple commit author information for your [commits](https://github.com/alenkacz/ci-signal-report/commits/master) in Alena's repository:
```
Max:
Max <16919345+mkorbi@users.noreply.github.com>
Max Körbächer <max@koerbaecher.io>

Leonard:
leonardpahlke <l.pahlke@reply.de>
leonardpahlke <leonardpahlke@icloud.com>
Leonard Vincent Simon Pahlke <58054077+leonardpahlke@users.noreply.github.com>
```

Do you prefer any specific author information that you want to be added as a Co-Author?

**Comment by palnabarun** [comment]
/hold
(putting an explicit hold for discussions)

**Comment by palnabarun** [comment]
To be precise, the linter script (`verify-golangci-lint`) is failing.

**Review by justaugustus — CHANGES_REQUESTED** [review]
@palnabarun -- I'm going to be a bit of a stickler here...

Can you instead migrate in @alenkacz's repo at https://github.com/alenkacz/ci-signal-report/tree/0ab843f97c5092081d5adaf9d50ec7f1c733a7b8?

With a new tool, we'd like to ensure thorough review of the content.
Several of the improvements to the tool since that commit have been merged in with minimal/no oversight of changes from experienced reviewers on the @kubernetes/release-engineering subproject.

Additionally, let's drop `ci-signal-reporter-approvers` until folks on that team have built up approval rights.

Explicit hold from @kubernetes/sig-release-admins.
/hold

**Comment by palnabarun** [comment]
@justaugustus -- Thank you for reviewing the changes!

> I'm going to be a bit of a stickler here...

Absolutely no issues! I raised this PR to figure what's best for the project and I appreciate the comments provided by you and anyone else. 👍🏽 

> Can you instead migrate in @alenkacz's repo at https://github.com/alenkacz/ci-signal-report/tree/0ab843f97c5092081d5adaf9d50ec7f1c733a7b8?

Done!

> With a new tool, we'd like to ensure thorough review of the content.
Several of the improvements to the tool since that commit have been merged in with minimal/no oversight of changes from experienced reviewers on the @kubernetes/release-engineering subproject.

Sounds good to me. The CI Signal Team can iterate over the initial seed of changes.

> Additionally, let's drop `ci-signal-reporter-approvers` until folks on that team have built up approval rights.

Done!

**Review by justaugustus — CHANGES_REQUESTED** [review]
@palnabarun -- Thanks for resolving my comments and extra kudos for the attribution work in the first commit! I think my only remaining blocking comment is to alpha-sort the owners alias.

@kubernetes/ci-signal -- I've left a set on non-blocking comments here that can be used as a guide for some follow-ups around code deduplication! :)

File: OWNERS_ALIASES
Comment (palnabarun) [reply OWNERS_ALIASES]: Done!

**Review by justaugustus — APPROVED** [review]
Great work, @palnabarun!
And thank you for the donation, @alenkacz 💕 

/hold cancel

File: cmd/ci-reporter/report.go
Comment (leonardpahlke) [reply cmd/ci-reporter/report.go:134]: tried, it doesn't seem to work well in this case because when you copy and paste the report into Slack, the formatting is lost (see screenshot).

slack-ci-signal-tablewriter-20211111191248-659x356
s

File: cmd/ci-reporter/report.go
Comment (leonardpahlke) [reply cmd/ci-reporter/report.go:65]: :+1:  #2314

File: cmd/ci-reporter/report.go
Comment (leonardpahlke) [reply cmd/ci-reporter/report.go:50]: #2314 updates this, release version is getting set over a flag now `-v`

File: cmd/ci-reporter/report.go
Comment (leonardpahlke) [reply cmd/ci-reporter/report.go:40]: interesting, will look into that later

**Pull request merged**
